spring-thrift-starter/src/main/java/info/developerblog/spring/thrift/client/ThriftClientBeanPostProcessor.java

```diff
@@ -109,6 +113,16 @@ public Object postProcessAfterInitialization(Object bean, String beanName) throw
         return bean;
     }
 
+    //We have to get a real bean in order to inject a thrift client into the bean instead of its proxy.
+    @SneakyThrows
+    private Object getTargetBean(Object bean) {
+        Object target = bean;
+        while (AopUtils.isAopProxy(target)) {
```

Comment:
added tests in new pull request: https://github.com/aatarasoff/spring-thrift-starter/pull/12